### ISSUE TITLE
POST body cannot contain more than one body parameter

### ISSUE THREAD

**Issue by colbroth** [abelsilva/swaggerwcf#113]
Swagger reports error "Operations must have no more than one body parameter". The swagger that is generated ends up having two parameters in the post request body and only one is allowed.

I added the following to the IStore.cs file

`        
[OperationContract]
[WebInvoke(Method = "POST")]
int BookTest(int id1, bool id2);
`

The swagger outputs the following:

`
"/v1/rest/BookTest":{"post":{"operationId":"SwaggerWcf.Test.Service.BookStore.BookTest","consumes":["application/json","application/xml"],"produces":["application/json","application/xml"],
"parameters":[
{"name":"id1","in":"body","required":true,"schema":{"type":"integer","format":"int32"}},{"name":"id2","in":"body","required":true,"schema":{"type":"boolean"}}],
"responses":{"default":{"description":"not available","schema":{"type":"integer","format":"int32"}}}}},
`

Expected:

`
 "/v1/rest/BookTest": {
      "post": {
        "operationId": "SwaggerWcf.Test.Service.BookStore.BookTest",
        "consumes": [
          "application/json",
          "application/xml"
        ],
        "produces": [
          "application/json",
          "application/xml"
        ],
        "parameters": [
          {
            "name": "BookTestBody",
            "in": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/SwaggerWcf.Test.Service.Data.BookTestBody"
            }
          }
        ],
        "responses": {
          "default": {
            "description": "not available",
            "schema": {
              "type": "integer",
              "format": "int32"
            }
          }
        }
      }
    },
`

with BookTestBody definition:

`
"SwaggerWcf.Test.Service.Data.BookTestBody": {
      "properties": {
        "id1": {
          "type": "integer",
          "format": "int32"
        },
        "id2": {
          "type": "integer",
          "format": "int32"
        }
      }
    }
`

**Comment by abelsilva** [comment]
can you try latest version?

**Comment by colbroth** [comment]
I tried with version 0.2.1

**Comment by abelsilva** [comment]
`0.2.2` is latest version

**Comment by colbroth** [comment]
This is now fixed in version 0.2.2

**Comment by JFDionne** [comment]
Hi, @abelsilva, this fix "broke" something else.
When we have a parameter path and a body on a post or a put, we got a "wrapper" on body parameter and the example value have a "data": {} but this is not true.

I'll suggest to do something like:
```
                // parameters - body
                int bodyParameterCount = 0;
                foreach (ParameterInfo parameter in parameters)
                {
                    if (GetInType(uriTemplate, parameter.Name) == InType.Body) bodyParameterCount++;
                }
```

and replace 
`if (!wrappedRequest && !isGetRequest && parameters.Length > 1)`
by
`if (!wrappedRequest && !isGetRequest && bodyParameterCount> 1)`
